### ISSUE TITLE
ResponsiveMenu broke dropdown menu after return to original breakpoint

### ISSUE THREAD

**Issue by AlexeySvidentsov** [zurb/foundation-sites#11611]
## What should happen?

I have Responsive menu: accordion on mobile and dropdown on desktop. At first initializing dropdown works fine. After changing viewport size to mobile menu converts to accordion and it works fine as well. But when i return to desktop/dropdown version - dropdown initialize seems to be failed. When i hover on item that has dropdown - it opens, and when i move mouse out of the item it should disappear.

## What happens instead?

Instead i've got console error, and submenu keep showing

`Uncaught TypeError: Cannot read property 'autoclose' of null`

foundation.dropdownMenu.js:150

## Possible Solution

...

## Test Case and/or Steps to Reproduce (for bugs)

Test Case: [codePen](https://codepen.io/iHectoR/pen/jQGWpB)

Foundation Docs for ResponsiveMenu has same issue - https://foundation.zurb.com/sites/docs/responsive-navigation.html#drilldown-dropdown-menu

## Your Environment

- Foundation version(s) used: 6.5.1
- Browser(s) name and version(s): Chrome, FireFox, Edge
- Device, Operating System and version: Windows 10
- Link to your project: 

## Checklist

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] There are no other issues similar to this one.
- [x] The issue title and template are correctly filled.

**Comment by AMarffy** [comment]
Hi Alexey! Your codepen actually works, since it's currently linked to Foundation 6.5.0's javascript.
(But there's still a console-error though.)
Maybe it should be linked to 6.5.1 for demonstration purposes.
I'm struggling with the exact same issue. Do you know if there's a patch for this? Thanks, Andrej

**Comment by AlexeySvidentsov** [comment]
Thank for correction. Replace with actual version.
For now, i've just cutted out **autclose** property, and the next **closingTime** property, since i'm not using it.
But still looking for the proper solution.

**Comment by AMarffy** [comment]
Thanks! I will try that too …

**Comment by Dureaghin** [comment]
Same issue. "hasSub is not defined" in foundation.dropdownMenu.js

Alexey, could you please provide your temp solution?

**Comment by DanielRuf** [comment]
Please test https://codepen.io/DanielRuf/pen/jJVjmz

**Comment by DanielRuf** [comment]
See https://github.com/zurb/foundation-sites/pull/11715

**Comment by Dureaghin** [comment]
Works for me! Thank you so much Daniel.

Alex.

**Comment by thejonroberts** [comment]
Just adding that I am seeing the same issue in 6.5.3 while trying to upgrade some packaged.  Daniel's PR fixes the issue.  Will have to wait on it.  Thanks Daniel.

**Comment by DanielRuf** [comment]
Hi @thejonroberts
You can use https://github.com/zurb/foundation-sites/pull/11715.patch or https://github.com/zurb/foundation-sites/pull/11715.diff with `patch-package` or `patch` to fix it in your projects.

There is currently no activity around Foundation Sites so the PR will not be merged so soon.

**Comment by thejonroberts** [comment]
I appreciate the heads up, Daniel.  I was considering using a fork, but ultimately decided to just disable the hover open behavior to work around this problem.

In case anyone else is looking for that solution, you can set `data-disable-hover=true, data-click-open=true` on the dropdown element, as described in the [foundation docs](https://foundation.zurb.com/sites/docs/dropdown-menu.html#js-options).

**Comment by Shirisu** [comment]
I hope the PR will be merged soon - we also run in this problem.
Disabling the hover isn't an option for us unfortunately.

**Comment by DanielRuf** [comment]
Hi @Shirisu,

the PR was already merged. It will be available in 6.6.0.

**Comment by Shirisu** [comment]
> Hi @Shirisu,
> 
> the PR was already merged. It will be available in 6.6.0.

Awesome, thanks for the information! :)

**Comment by kieran737** [comment]
Hi,

I have found a similar issue regarding the dropdownMenu,js where auto close is failing to function after a breakpoint change similar to the issue reported on this post: https://github.com/foundation/foundation-sites/issues/11611 and I believe the fix was merged for release 6.6.0 and I am using 6.6.1, on Chrome 88.0.4324.150 (Official Build) (32-bit).

When I have the menu configured to display the dropdown layout for desktop and then accordion layout for mobile and I switch breakpoints just by adjusting the viewport size via maximise & minimise the auto close functionality seems to fail, when I re-initialise the menu either by refreshing the page or via the JS the problem goes away.

Here is a codepen I have created to isolate the issue: https://codepen.io/k-randl3man/pen/BaQZRBG

How to reproduce:
1. Maximise window.
2. Minimise Window.
3. Hover over products and choose and sub menu.
4. Move mouse outside of the menu.
